src/DepartmentService.Business/User/CreateDepartmentUsersCommand.cs

```diff
@@ -44,8 +46,10 @@ public class CreateDepartmentUsersCommand : ICreateDepartmentUsersCommand
 
     public async Task<OperationResultResponse<bool>> ExecuteAsync(Guid departmentId, List<Guid> usersIds)
     {
-      if (!await _accessValidator.HasRightsAsync(Rights.EditDepartmentUsers) &&
-        !await _accessValidator.HasRightsAsync(Rights.AddEditRemoveDepartments))
+      DbDepartmentUser sender = await _repository.GetAsync(_httpContextAccessor.HttpContext.GetUserId());
+
+      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveDepartments) &&
+        !(await _accessValidator.HasRightsAsync(Rights.EditDepartmentUsers) && sender != null && sender.DepartmentId == departmentId))
```

Comment:
i think, it will be better if you get the sender here, it will easy do not get data from DB if user is Rights.AddEditRemoveDepartments.
do you need to check sender != null?  you can write sender?.DepartmentId